packages/block-editor/src/hooks/style.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * External dependencies
+ */
+import { mapKeys, kebabCase, isObject, entries } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { hasBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { COLOR_SUPPORT_KEY } from './color';
+
+const styleSupportKeys = [ COLOR_SUPPORT_KEY ];
+
+const hasStyleSupport = ( blockType ) =>
+	styleSupportKeys.some( ( key ) => hasBlockSupport( blockType, key ) );
+
+/**
+ * Flatten a nested Global styles config and generates the corresponding
+ * flattened CSS variables.
+ *
+ * @param  {Object} styles Styles configuration
+ * @return {Object}        Flattened CSS variables declaration
+ */
+function getCSSVariables( styles = {} ) {
+	const prefix = '--wp';
+	const token = '--';
+	const getNestedCSSVariables = ( config ) => {
+		let result = {};
+		entries( config ).forEach( ( [ key, value ] ) => {
+			if ( ! isObject( value ) ) {
+				result[ key ] = value;
+				return;
+			}
+
+			result = {
+				...result,
+				...mapKeys(
+					getNestedCSSVariables( value ),
+					( _, subkey ) => kebabCase( key ) + token + subkey
+				),
+			};
+		} );
+
+		return result;
+	};
+
+	return mapKeys(
+		getNestedCSSVariables( styles ),
+		( _, key ) => prefix + token + key
+	);
+}
+
+/**
+ * Filters registered block settings, extending attributes to include `style` attribute.
+ *
+ * @param  {Object} settings Original block settings
+ * @return {Object}          Filtered block settings
+ */
+function addAttribute( settings ) {
+	if ( ! hasStyleSupport( settings ) ) {
+		return settings;
+	}
+
+	// allow blocks to specify their own attribute definition with default values if needed.
+	if ( ! settings.attributes.style ) {
+		Object.assign( settings.attributes, {
+			style: {
+				type: 'object',
+			},
+		} );
+	}
+
+	return settings;
+}
+
+/**
+ * Override props assigned to save component to inject colors classnames and styles.
```

Comment:
It looks like it’s only styles. I don’t see operations on class names.

Reply:
Yes, I splitted into two hooks.

Reply:
The comment needs to get updated then.